backends/postgres/postgres_backend_test.go

```diff
@@ -404,22 +374,72 @@ func TestCron(t *testing.T) {
 	if err != nil {
 		t.Error(err)
 	}
+}
 
-	t.Cleanup(func() {
-		flushDB()
-	})
+func TestMultipleCronNodes(t *testing.T) {
+	jobsProcessed := sync.Map{}
+	const numberOfWorkers = 4
+	const cron = "* * * * * *"
+	connString, _ := prepareAndCleanupDB(t)
+
+	workers := make([]neoq.Neoq, numberOfWorkers)
+	var jobsCompleted uint32
+	var duplicateJobs uint32
+	for i := 0; i < numberOfWorkers; i++ {
+		ctx := context.TODO()
+		nq, err := neoq.New(ctx, neoq.WithBackend(postgres.Backend), postgres.WithConnectionString(connString))
+		if err != nil {
+			t.Fatal(err)
+		}
+		t.Cleanup(func() {
+			nq.Shutdown(ctx)
+		})
+		h := handler.NewPeriodic(func(ctx context.Context) (err error) {
+			job, err := jobs.FromContext(ctx)
+			if err != nil {
+				t.Fatalf("failed to extract job details from context: %+v", err)
+				return nil
+			}
+			_, exists := jobsProcessed.LoadOrStore(job.ID, "foo")
+			if exists {
+				t.Fatalf("job (%d) has already been processed by another worker!", job.ID)
+				atomic.AddUint32(&duplicateJobs, 1)
+				return nil
+			}
+			atomic.AddUint32(&jobsCompleted, 1)
+			return
+		})
+
+		h.WithOptions(
+			handler.JobTimeout(500*time.Millisecond),
+			handler.Concurrency(1),
+		)
+
+		err = nq.StartCron(ctx, cron, h)
+		if err != nil {
+			t.Error(err)
+		}
+
+		workers[i] = nq
+	}
+
+	// allow time for listener to start and for at least one job to process
+	time.Sleep(2 * time.Second)
+	if jobsCompleted == 0 {
+		t.Fatalf("no jobs were completed after 2 seconds?")
```

Comment:
See above comment, and then we can change this to:

`t.Fatalf("no jobs were completed after %v", SlightlyMoreThanOneSecond)`

Reply:
Updated!